### PR TITLE
Fix: Remove the 20 chars reference from doc block comments of `the_password_form` hook.

### PR DESCRIPTION
**Pull request by Debarghya-Banerjee**
Trac Ticket: [Core-61703](https://core.trac.wordpress.org/ticket/61703)

## Problem Statement:

- Beginning with WordPress 3.6, passwords are no longer stored as plaintext but instead as hashed values. This change eliminates the previous length limitation imposed by plaintext passwords.

## Solution:

- Removed the comment that had the 20 chars reference from the Doc Block comment.

### CODE REVIEW

**Review by mukeshpanchal27 — APPROVED** [review]

**Comment by mukeshpanchal27** [comment]
Thanks @Debarghya-Banerjee, PR LGTM!

**Comment by Debarghya-Banerjee** [comment]
Hi @peterwilsoncc, I have made the necessary changes, can you please check once. Sorry for the delay.

File: src/wp-includes/post-template.php [hunk 1]
```diff
@@ -1771,14 +1771,14 @@ function get_the_password_form( $post = 0 ) {
 	$label  = 'pwbox-' . ( empty( $post->ID ) ? rand() : $post->ID );
 	$output = '<form action="' . esc_url( site_url( 'wp-login.php?action=postpass', 'login_post' ) ) . '" class="post-password-form" method="post">
 	<p>' . __( 'This content is password protected. To view it please enter your password below:' ) . '</p>
-	<p><label for="' . $label . '">' . __( 'Password:' ) . ' <input name="post_password" id="' . $label . '" type="password" spellcheck="false" size="20" /></label> <input type="submit" name="Submit" value="' . esc_attr_x( 'Enter', 'post password form' ) . '" /></p></form>
+	<p><label for="' . $label . '">' . __( 'Password:' ) . ' <input name="post_password" id="' . $label . '" type="password" spellcheck="false" size="255" /></label> <input type="submit" name="Submit" value="' . esc_attr_x( 'Enter', 'post password form' ) . '" /></p></form>
```
Comment (peterwilsoncc) [inline src/wp-includes/post-template.php]:
```suggestion
	<p><label for="' . $label . '">' . __( 'Password:' ) . ' <input name="post_password" id="' . $label . '" type="password" spellcheck="false" size="20" /></label> <input type="submit" name="Submit" value="' . esc_attr_x( 'Enter', 'post password form' ) . '" /></p></form>
```

File: src/wp-includes/post-template.php [hunk 2]
```diff
@@ -1771,14 +1771,14 @@ function get_the_password_form( $post = 0 ) {
...
 	';
 
 	/**
 	 * Filters the HTML output for the protected post password form.
 	 *
 	 * If modifying the password field, please note that the core database schema
```
Comment (peterwilsoncc) [inline src/wp-includes/post-template.php]:
```suggestion
	 * If modifying the password field, please note that the WordPress database schema
```

File: src/wp-includes/post-template.php [hunk 3]
```diff
@@ -1771,14 +1771,14 @@ function get_the_password_form( $post = 0 ) {
...
-	 * limits the password field to 20 characters regardless of the value of the
+	 * limits the password field to 255 characters regardless of the value of the
 	 * size attribute in the form input.
```
Comment (peterwilsoncc) [inline src/wp-includes/post-template.php]:
```suggestion
	 * limits the password length to 255 characters regardless of the value of any
	 * `minlength` or `maxlength` attributes or other validation that may be added to
	 * the input.
```

**Review by peterwilsoncc — COMMENTED** [review]
Sorry, I realised I was confusing the `size` an `maxlength` attributes. My earlier suggestion was incorrect and causes horizontal scroll  

Screenshot 2024-08-02 at 8 58 44 AM

Hopefully this will be the last change request to the PR, thanks for being patient as I figured what actually needed to change.

File: src/wp-includes/post-template.php (hunk 2 above)
Comment (Debarghya-Banerjee) [reply src/wp-includes/post-template.php]: Fixed

File: src/wp-includes/post-template.php (hunk 3 above)
Comment (Debarghya-Banerjee) [reply src/wp-includes/post-template.php]: Added

File: src/wp-includes/post-template.php (hunk 1 above)
Comment (Debarghya-Banerjee) [reply src/wp-includes/post-template.php]: Fixed

**Comment by Debarghya-Banerjee** [comment]
Hi @peterwilsoncc, I have addressed these changes. Can you please take a look into it? Thanks.

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r58846](https://core.trac.wordpress.org/changeset/58846).

**Pull request closed without merging**